### PR TITLE
cmake: Build `libbitcoinkernel` and `bitcoin-chainstate`

### PR DESCRIPTION
**Pull request by hebasto**
New build configuration options:
- `-DBUILD_KERNEL_LIB`
- `-DBUILD_UTIL_CHAINSTATE`

To build a shared `libbitcoinkernel`, one can specify `-DBUILD_SHARED_LIBS=ON`.

### CODE REVIEW

**Comment by hebasto** [comment]
Friendly ping @TheCharlatan @fanquake @theuni @vasild @m3dwards @pablomartin4btc :)

File: src/kernel/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+# libbitcoinkernel requires default symbol visibility, explicitly
+# specify that here so that things still work even when user
+# configures with -DREDUCE_EXPORTS=ON
+#
+# Note this is a quick hack that will be removed as we
+# incrementally define what to export from the library.
+set_target_properties(bitcoinkernel PROPERTIES
+  CXX_VISIBILITY_PRESET default
+)
```
Comment (TheCharlatan) [inline src/kernel/CMakeLists.txt:108]: We currently copy the library to an install directory when running `make install`, but it seems to be missing in the install dir here.

```
diff --git a/src/kernel/CMakeLists.txt b/src/kernel/CMakeLists.txt
index cc0263ce3c..620a31134d 100644
--- a/src/kernel/CMakeLists.txt
+++ b/src/kernel/CMakeLists.txt
@@ -108,0 +109,4 @@ set_target_properties(bitcoinkernel PROPERTIES
+
+install(TARGETS bitcoinkernel
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
```

Comment (hebasto) [reply src/kernel/CMakeLists.txt:108]: Thanks! Added.

**Review by TheCharlatan — APPROVED** [review]
lgtm

**Review by pablomartin4btc — APPROVED** [review]
tACK f9dc316d9311da89d3636ff2dd5834f00889fae0

<details>
<summary>Tested on Ubuntu 22.04.</summary>

```
...

Configure summary
=================
Executables:

...


  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
```


</details>

**Pull request merged**